scrape/scrape.go

```diff
@@ -166,72 +198,115 @@ func newScrapePool(cfg *config.ScrapeConfig, app storage.Appendable, offsetSeed
-			ctx,
-			opts.scraper,
-			logger.With("target", opts.target),
-			buffers,
-			func(l labels.Labels) labels.Labels {
-				return mutateSampleLabels(l, opts.target, opts.honorLabels, opts.mrc)
+		activeTargets:        map[uint64]*Target{},
+		scrapeLoopOptions: scrapeLoopOptions{
+			appendable:  appendable,
+			buffers:     buffers,
+			offsetSeed:  offsetSeed,
+			symbolTable: symbols,
+			metrics:     metrics,
+
+			// config.ScrapeConfig.
+			sampleLimit: int(cfg.SampleLimit),
+			bucketLimit: int(cfg.NativeHistogramBucketLimit),
+			maxSchema:   pickSchema(cfg.NativeHistogramMinBucketFactor),
+			labelLimits: &labelLimits{
+				labelLimit:            int(cfg.LabelLimit),
+				labelNameLengthLimit:  int(cfg.LabelNameLengthLimit),
+				labelValueLengthLimit: int(cfg.LabelValueLengthLimit),
 			},
-			func(l labels.Labels) labels.Labels { return mutateReportSampleLabels(l, opts.target) },
-			func(ctx context.Context) storage.Appender { return app.Appender(ctx) },
-			cache,
-			sp.symbolTable,
-			offsetSeed,
-			opts.honorTimestamps,
-			opts.trackTimestampsStaleness,
-			opts.enableCompression,
-			opts.sampleLimit,
-			opts.bucketLimit,
-			opts.maxSchema,
-			opts.labelLimits,
-			opts.interval,
-			opts.timeout,
-			opts.alwaysScrapeClassicHist,
-			opts.convertClassicHistToNHCB,
-			cfg.ScrapeNativeHistogramsEnabled(),
-			options.EnableStartTimestampZeroIngestion,
-			options.EnableTypeAndUnitLabels,
-			options.ExtraMetrics,
-			options.AppendMetadata,
-			opts.target,
-			options.PassMetadataInContext,
-			metrics,
-			options.skipOffsetting,
-			sp.validationScheme,
-			sp.escapingScheme,
-			opts.fallbackScrapeProtocol,
-		)
+			honorLabels:                   cfg.HonorLabels,
+			honorTimestamps:               cfg.HonorTimestamps,
+			trackTimestampsStaleness:      cfg.TrackTimestampsStaleness,
+			enableNativeHistogramScraping: cfg.ScrapeNativeHistogramsEnabled(),
+			alwaysScrapeClassicHist:       cfg.AlwaysScrapeClassicHistogramsEnabled(),
+			convertClassicHistToNHCB:      cfg.ConvertClassicHistogramsToNHCBEnabled(),
+			fallbackScrapeProtocol:        cfg.ScrapeFallbackProtocol.HeaderMediaType(),
+			enableCompression:             cfg.EnableCompression,
+			mrc:                           cfg.MetricRelabelConfigs,
+			validationScheme:              cfg.MetricNameValidationScheme,
+			escapingScheme:                escapingScheme,
+
+			// scrape.Options.
+			enableSTZeroIngestion:   options.EnableStartTimestampZeroIngestion,
+			enableTypeAndUnitLabels: options.EnableTypeAndUnitLabels,
+			reportExtraMetrics:      options.ExtraMetrics,
+			appendMetadataToWAL:     options.AppendMetadata,
+			passMetadataInContext:   options.PassMetadataInContext,
+			skipOffsetting:          options.skipOffsetting,
+		},
 	}
 	sp.metrics.targetScrapePoolTargetLimit.WithLabelValues(sp.config.JobName).Set(float64(sp.config.TargetLimit))
 	return sp, nil
 }
 
+// newScrapeLoop constructs new scrapeLoop.
+func newScrapeLoop(
+	sp *scrapePool,
```

Comment:
Why isn't this a function on `(sp *scrapePool) ` ? All options for the new pool come from the scapePool.
```suggestion
func (sp *scrapePool)  newScrapeLoop(
```